pilot/pkg/xds/xdsgen.go

```diff
@@ -187,3 +188,45 @@ func ResourceSize(r model.Resources) int {
 	}
 	return size
 }
+
+// xdsNeedsPush checks for the common cases whether we need to push or not.
+func xdsNeedsPush(req *model.PushRequest, proxy *model.Proxy, allowIncrementalPush bool) (bool, bool) {
```

Comment:
we donot need to add `allowIncrementalPush` here, confusing actually not means incremental as what i understand for eds.  we can do that value check in each specific needPush function, that' more clear